model/market.ts

```diff
@@ -233,9 +246,50 @@ export const calculateTotalDeposit = (
         initialDeposit,
       ))
     }
-    initialDeposit = markets.reduce((acc, market, i) => acc + amountOuts[i], 0n)
+    initialDeposit = amountOuts.reduce((a, b) => a + b, 0n)
     totalDeposit = totalDeposit + initialDeposit
   }
 
   return totalDeposit
 }
+
+export const calculateDepositApy = (
+  substitute: Currency,
+  markets: Market[],
+  initialDeposit: bigint,
+  currentTimestamp: number,
+): {
+  epochStart: Date
+  epochEnd: Date
+  proceeds: bigint
+  apy: number
+} => {
+  if (
```

Comment:
`some` looks more fitting in this situation.
```
    markets.some(
      (market) =>
        !isAddressEqual(
          market.quoteToken.address,
          substitute.address as `0x${string}`,
        ),
    )
```

Reply:
https://github.com/clober-dex/coupon.finance/pull/25/commits/a22bdefe07066d04a9528243ec82943f136f5239